workshop/test.rb

```diff
@@ -0,0 +1,119 @@
+        obj = Framework.new()
+        obj.opening_site_and_maximizing(url)
+        submit_btn = obj.get_element_by_tag_name("button")
+        obj.clicking_element(submit_btn)
+        obj.wait_for(2)
+        name_input_tag = obj.get_element_by_id("name")
+        assert_equal("",name_input_tag["value"])
+        obj.quit_browser()
+    end  
+
+    def test_get_element_by_id
+        # url = "file:///C:/Users/dines/git/infuse/Practice-Dinesh/workshop/index.html"
+        url = "file:///app/index.html"
+        obj = Framework.new()
+        obj.opening_site_and_maximizing(url)
+        element = obj.get_element_by_id("name")
+        assert_equal("name",element['id'])
+        obj.quit_browser()
+    end 
+
+    def test_get_element_by_id_negativetest
+        # url = "file:///C:/Users/dines/git/infuse/Practice-Dinesh/workshop/index.html"
+        url = "file:///app/index.html"
+        obj = Framework.new()
+        obj.opening_site_and_maximizing(url)
+        element = obj.get_element_by_id("nameee")
+        assert_not_equal("name",element['id'])
+        obj.quit_browser()
+    end 
+
+    def test_get_element_by_class
+        # url = "file:///C:/Users/dines/git/infuse/Practice-Dinesh/workshop/index.html"
+        url = "file:///app/index.html"
+        obj = Framework.new()
+        obj.opening_site_and_maximizing(url)
+        element = obj.get_element_by_class("number")
+        assert_equal("number",element['class'])
+        obj.quit_browser()
+    end
+
+    def test_get_element_by_class_negativetest
+        # url = "file:///C:/Users/dines/git/infuse/Practice-Dinesh/workshop/index.html"
+        url = "file:///app/index.html"
+        obj = Framework.new()
+        obj.opening_site_and_maximizing(url)
+        element = obj.get_element_by_class("numberrr")
+        assert_not_equal("number",element['class'])
+        obj.quit_browser()
+    end
+
+    def test_get_elements_by_class
+        # url = "file:///C:/Users/dines/git/infuse/Practice-Dinesh/workshop/index.html"
+        url = "file:///app/index.html"
+        obj = Framework.new()
+        obj.opening_site_and_maximizing(url)
+        element = obj.get_elements_by_class("number")[1]
+        assert_equal("2",element.text)
+        obj.quit_browser()
+    end
+
+    def test_get_elements_by_class_negativetest
+        # url = "file:///C:/Users/dines/git/infuse/Practice-Dinesh/workshop/index.html"
+        url = "file:///app/index.html"
+        obj = Framework.new()
+        obj.opening_site_and_maximizing(url)
+        element = obj.get_elements_by_class("numberrr")
+        assert_not_equal("2",element)
+        obj.quit_browser()
+    end
+
+    
+    def test_get_element_by_tag_name_negativetest
+        # url = "file:///C:/Users/dines/git/infuse/Practice-Dinesh/workshop/index.html"
+        url = "file:///app/index.html"
+        obj = Framework.new()
+        obj.opening_site_and_maximizing(url)
+        element = obj.get_element_by_tag_name("inputtt")
```

Comment:
Each unit test should test only **one** function. This unit test uses the `get_element_by_tag_name` method.

Reply:
You can use the driver methods to get the text

Reply:
Change other unit tests similarly